docs/architecture/adr-047-handling-evidence-from-light-client.md

```diff
@@ -0,0 +1,121 @@
+# ADR 047: Handling evidence from light client
+
+## Changelog
+* 18-02-2020: Initial draft
+
+## Context
+
+If the light client is under attack (either directly - lunatic / phantom
+validators or indirectly - fork on the main chain), it's supposed to halt and
+send an evidence of misbehavior to the correct full node. Upon receiving an
+evidence, full node should punish malicious validators (if possible).
+
+## Decision
+
+When a light client sees two conflicting headers (`H1.Hash() != H2.Hash()`,
+`H1.Height == H2.Height`), both having 1/3+ of voting power of the currently
+trusted validator set, it will submit a `ConflictingHeadersEvidence` to all
+full nodes it's connected to.
+
+```go
+type ConflictingHeadersEvidence struct {
+  H1 *types.SignedHeader
+  H2 *types.SignedHeader
+}
+```
+
+When a full node receives the `ConflictingHeadersEvidence` evidence, it should
+a) validate it b) figure out if malicious behaviour is obvious (immediately
+slashable) or the fork accountability protocol needs to be started.
+
+(a): check both headers are valid (`ValidateBasic`), have
+the same height and signed by 1/3+ of known validator set.
+
+- What to do if neither H1 nor H2 was committed on the main chain?
```

Comment:
Is that possible if both headers are each signed by more than 1/3 of the known validator set and there are no double signings

Reply:
yes, see https://github.com/tendermint/tendermint/issues/4182